eth/api.go

```diff
@@ -812,3 +813,23 @@ func (api *ScrollAPI) GetSkippedTransactionHashes(ctx context.Context, from uint
 
 	return hashes, nil
 }
+
+// CalculateRowConsumptionByBlockNumber
+func (api *ScrollAPI) CalculateRowConsumptionByBlockNumber(ctx context.Context, number rpc.BlockNumber) (*types.RowConsumption, error) {
```

Comment:
If we call this with a non-canonical (forked) block, it will still calculate ccc but it won't write it to db (so it will return `null`), is that correct?

Reply:
It is not possible to call this for a non-canonical block as `number` only indexes in to the canonical chain.

Reply:
True, but what happens in the unlikely scenario that we call this API and the block gets reorged after the `GetBlockByNumber` call?

Reply:
As long as the now-sidechain that that block belongs to is still available, we will run CCC on it and store the result in the database. It doesn't really matter for AsyncChecker if a block is canonical or not.